Mathlib/CategoryTheory/Presentable/OrthogonalReflection.lean

```diff
@@ -0,0 +1,230 @@
+/-
+Copyright (c) 2025 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.CategoryTheory.Limits.Shapes.Multiequalizer
+import Mathlib.CategoryTheory.Localization.Bousfield
+import Mathlib.CategoryTheory.SmallObject.Iteration.Basic
+
+/-!
+# The Orthogonal-reflection construction
+
+Given `W : MorphismProperty C` (which should be small) and assuming the existence
+of certain colimits in `C`, we construct a morphism `toSucc W Z : Z ⟶ succ W Z` for
+any `Z : C`. This morphism belongs to `LeftBousfield.W W.isLocal` and
+is an isomorphism iff `Z` belongs to `W.isLocal`.
+
+## References
+* [Adámek, J. and Rosický, J., *Locally presentable and accessible categories*][Adamek_Rosicky_1994]
```

Comment:
Would you mind giving more specific references where possible?

(I am inclined to simply assume that the definition here faithfully reflects what is in the book, but I wouldn't mind being able to quickly check. :-)

Reply:
I have added more details in the docstring of the file.